### ISSUE TITLE
Can't install extensions

### ISSUE THREAD

**Issue by FallBackITA27** [zed-industries/zed#14386]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Read title. This is basically a duplicate of https://github.com/zed-industries/zed/issues/10330, but the logs are different

### Environment

Fedora 40.

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

2024-07-13T09:52:20+02:00 [INFO] installing extension html latest version
2024-07-13T09:52:26+02:00 [WARN] request completed with error: request or operation took longer than the configured timeout time
2024-07-13T09:52:26+02:00 [ERROR] error downloading extension: request or operation took longer than the configured timeout time
</pre></details>

**Comment by jaeyson** [comment]
one workaround to install extension:

## Finding the extension link

an example link: `https://api.zed.dev/extensions/<EXTENSION_ID>/download`

where to get `EXTENSION_ID`?
1. go to extensions
2. find the extension you want to install
3. on the right side you can see its button "Install" and the "Github icon"
![Screenshot from 2024-07-12 17-58-33](https://github.com/user-attachments/assets/04295711-b10f-442e-b43a-67dd2f8c03b0)
4. click the Github icon so it'll go to the github page
5. open the `extension.toml`
6. look for the `id`, in this example:
```ini
id = "macos-classic"   <--- this ID
name = "macOS Classic Theme"
description = "A macOS native style theme, let it same like native app in macOS."
version = "0.0.8"
schema_version = 1
authors = ["Jason Lee <huacnlee@gmail.com>"]
repository = "https://github.com/huacnlee/zed-theme-macos-classic"
```

so the link will be: `https://api.zed.dev/extensions/macos-classic/download`

## Download the extension using a disposable browser or another way
You can use vpn or any as long as you can download the said link

OR

Visit this site: https://sqrx.com, then click `personal`. In this step you can use e.g. Brave or Chrome (I think Firefox not supported on this site). From there you can download the link

## Where to save the extension after downloading it
- Close Zed editor
- extract the `archive.tar.gz` file and rename the extracted folder according to the extension id
- copy the extracted folder on these path:
  - osx: `~/Library/Application\ Support/Zed/extensions/installed/extracted_folder_name`
  - ubuntu: `~/.local/share/zed/extensions/installed/extracted_folder_name`

Then open Zed, check if installed. Lmk if this works.

**Comment by FallBackITA27** [comment]
Thanks, even though I had preceded you. It does work.

**Comment by notpeter** [comment]
I'm going to go ahead and close this as unactionable.

For the moment we are using Digital Ocean Spaces (S3 compat) for storage of Extension artifacts.  This service appears to be blocked by some networks (considered Malware).  Perhaps we will migrate to another service at some point in the future in attempt to fix this, but convincing random network operators (mostly foreign) to stop using a block list of unknown origin is out of scope at this time.

If you or anyone has any leads about the source of these IP-block list, I would be happy to investigate further.

Thanks for reporting.
